### PR TITLE
Feature#71 only the user can edit and delete their own coupons

### PR DESCRIPTION
**Pull request by DiegoCoronaBarragan**
# Description

Implement that only the user can edit and delete their own coupons and edit styles.

## Screenshots

image

### CODE REVIEW

**Review by AangelF — APPROVED** [review]

File: app/views/mainpages/_content.html.erb [hunk 1]
```diff
@@ -21,14 +21,10 @@
               <p><%= result.description %></p>
             </div>
             <div class="coupon_opc">
-              <%= button_to "delete", {:controller => :cupons, :action => 'destroy', :id => result.id }, :method => :delete %>
-              <%= link_to "Edit", edit_cupon_path(result)%>
+              <%= link_to "Ver", cupon_path(result.id) %>
+              <%= ((button_to "delete", {:controller => :cupons, :action => 'destroy', :id => result.id }, :method => :delete) if current_user.id == result.user_id) if current_user %>
```
Comment (juperameza) [inline app/views/mainpages/_content.html.erb:25]: The :method works? As long as I remember it doesn't

File: app/views/mainpages/_content.html.erb [hunk 2]
```diff
@@ -21,14 +21,10 @@
...
+              <%= ((link_to "Edit", edit_cupon_path(result)) if current_user.id == result.user_id) if current_user %>
```
Comment (juperameza) [inline app/views/mainpages/_content.html.erb:26]: Also I think that's a lot of nested ifs in the view I recommend trying to charge the controller with the logic

**Review by juperameza — CHANGES_REQUESTED** [review]

File: app/views/mainpages/_content.html.erb (hunk 1 above)
Comment (DiegoCoronaBarragan) [reply app/views/mainpages/_content.html.erb]: The :method does work, we have tested it and everything works.

File: app/views/mainpages/_content.html.erb (hunk 2 above)
Comment (DiegoCoronaBarragan) [reply app/views/mainpages/_content.html.erb]: I reduced it as much as possible but if I modify it more for some reason the delete button disappears, that's the only way it renders and works correctly. :(

File: app/views/mainpages/_content.html.erb [hunk 3]
```diff
@@ -21,14 +21,10 @@
               <p><%= result.description %></p>
             </div>
             <div class="coupon_opc">
-              <%= button_to "delete", {:controller => :cupons, :action => 'destroy', :id => result.id }, :method => :delete %>
-              <%= link_to "Edit", edit_cupon_path(result)%>
+              <%= link_to "Ver", cupon_path(result.id) %>
+              <%= (button_to "delete", {:controller => :cupons, :action => 'destroy', :id => result.id }, :method => :delete) if current_user.id == result.user_id && current_user %>
```
Comment (YilKanda) [inline app/views/mainpages/_content.html.erb:25]:
```suggestion
              <%= (button_to "delete", {controller: :cupons, action: 'destroy', id: result.id }, method: :delete) if current_user.id == result.user_id && current_user %>
```

Comment (YilKanda) [inline app/views/mainpages/_content.html.erb:25]:
```suggestion
              <%= (button_to "delete", {controller: :cupons, action: 'destroy', id: result.id }, method: :delete) if current_user.id == result.user_id && current_user %>
```

**Review by YilKanda — APPROVED** [review]

**Review by EverardoAlcarazBaltazar — APPROVED** [review]

**Review by juperameza — APPROVED** [review]
Lgtm

**Pull request merged**